pallets/xor-fee/src/lib.rs

```diff
@@ -695,7 +733,40 @@ pub mod pallet {
     pub struct Pallet<T>(PhantomData<T>);
 
     #[pallet::hooks]
-    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        #[cfg(feature = "ready-to-test")] // Dynamic fee
+        fn on_idle(current_block: BlockNumberFor<T>, _remaining_weight: Weight) -> Weight {
+            let mut weight: Weight = Weight::default();
+            if let Some(next_update_block) = Self::next_update_block() {
+                if current_block >= next_update_block {
```

Comment:
You have to check that `BlocksToUpdate != 0`, because if it is 0 - it means the tech committee turned off the dynamic update.
But now, if the value is 0, the hook will trigger each block because `next_block_number == current_block`

Reply:
Or, as @vovac12 mentions above, you have to check that period is not `None` if it will be an `Option`.
However, you still need to check zero case. And add min possible period check probably.

Reply:
If 0 is not an off number, we also must check it not be set via extrinsic.

Reply:
> And add min possible period check probably.

Maybe max too?

Reply:
Maybe. It's necessary to clarify requirements.

Reply:
I think we can just turn off on 0, so min check is not necessary. And I don't think that we need to check for the max value